zwavequbino/integrationpluginzwavequbino.cpp

```diff
@@ -0,0 +1,185 @@
+        info->finish(Thing::ThingErrorHardwareNotAvailable);
+        return;
+    }
+
+    ZWaveNode *node = nodeForThing(thing);
+
+    if (thing->thingClassId() == flushShutterThingClassId) {
+
+        thing->setStateValue(flushShutterPercentageStateTypeId, node->value(ZWaveValue::GenreUser, ZWaveValue::CommandClassSwitchMultilevel, 1, 0, ZWaveValue::TypeByte).value().toUInt() * 100 / 255);
+
+        info->finish(Thing::ThingErrorNoError);
+
+        connect(node, &ZWaveNode::valueChanged, thing, [thing](const ZWaveValue &value){
+            if (value.genre() == ZWaveValue::GenreConfig && value.commandClass() == ZWaveValue::CommandClassConfiguration && value.index() == 74) {
+                qCDebug(dcZWaveQubino()) << "Open/Close time value changed:" << value.value();
+                thing->setSettingValue(flushShutterSettingsOpenCloseTimeParamTypeId, value.value());
+            }
+
+            else if (value.genre() == ZWaveValue::GenreUser && value.commandClass() == ZWaveValue::CommandClassSwitchMultilevel && value.index() == 0) {
+                qCDebug(dcZWaveQubino()) << "Level value changed:" << value.value();
+                thing->setStateValue(flushShutterPercentageStateTypeId, 100 - value.value().toUInt());
+            }
+
+            else {
+                qCWarning(dcZWaveQubino()) << "Unhandled value changed" << value;
+            }
+        });
+
+        connect(info->thing(), &Thing::settingChanged, node, [node](const ParamTypeId &settingId, const QVariant &value){
+            qCDebug(dcZWaveQubino()) << "Setting qubino settings!!!";
+            if (settingId == flushShutterSettingsOpenCloseTimeParamTypeId) {
+                ZWaveValue openCloseDurationValue = node->value(ZWaveValue::GenreConfig, ZWaveValue::CommandClassConfiguration, 1, 74, ZWaveValue::TypeShort);
+                openCloseDurationValue.setValue(value);
+                node->setValue(openCloseDurationValue);
+            }
+        });
+
+        return;
+    }
+
+    info->finish(Thing::ThingErrorNoError);
+}
+
+void IntegrationPluginZWaveQubino::executeAction(ThingActionInfo *info)
+{
+    if (!hardwareManager()->zwaveResource()->available()) {
+        info->finish(Thing::ThingErrorHardwareNotAvailable);
+        return;
+    }
+
+    Thing *thing = info->thing();
+    ZWaveNode *node = nodeForThing(info->thing());
+    if (!node->reachable()) {
+        info->finish(Thing::ThingErrorHardwareNotAvailable);
+        return;
+    }
+
+    if (thing->thingClassId() == flushShutterThingClassId) {
+        if (info->action().actionTypeId() == flushShutterOpenActionTypeId) {
+            ZWaveValue powerValue = node->value(ZWaveValue::GenreUser, ZWaveValue::CommandClassSwitchMultilevel, 1, 7, ZWaveValue::TypeButton);
+            powerValue.setValue(true);
+            node->setValue(powerValue);
+//            thing->setStateValue(powerSocketPowerStateTypeId, info->action().paramValue(powerSocketPowerActionPowerParamTypeId));
```

Comment:
sure you don't want to update the state here? and in the other actions? otherwise remove the comment?

Reply:
yep. copy/paste leftover from a setPower, but this is open/close actions etc... they don't immediately update a state.